### ISSUE TITLE
filter analytical mean zeros from summarycalc

### ISSUE THREAD

**Issue by johcarter** [OasisLMF/ktools#41]
1. Filter records with zero loss including sidx -1 from summarycalc  
2. Retain output of all zero losses in summarycalc as debug option with command line parameter -z.
3. Review filtering logic of gulcalc and fmcalc and make consistent.

Currently
* gulcalc accepts a loss threshold -L {} and filters out all samples for sidx>0 below that threshold
* fmcalc filters out all samples with zero loss for sidx > 0 automatically
* summarycalc does no filtering.

Suggest the following;
1) All three components filter out samples with loss = 0 by default including sidx -1
2) All components have the same debug option to output all zero losses
3) gulcalc additionally filters out losses < x when -L x is used
4) Add the -L parameter to both fmcalc and summarycalc to make symmetric with gulcalc. the python wrapper can interpret how to apply user input loss threshold.

1 is quite radical and may 'break' things if we filter out sidx -1 because the stream structure might change. if too radical, we could only filter zero loss for sidx>0. but lets see.

**Comment by johcarter** [comment]
summarycalc filters out zero losses by default, with -z as a debug option
no change to gulcalc
no change to fmcalc.
